eo-runtime/src/main/eo/org/eolang/switch.eo

```diff
@@ -0,0 +1,63 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2016-2022 Yegor Bugayenko
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
++package org.eolang
++home https://github.com/objectionary/eo
++version 0.0.0
++rt jvm org.eolang:eo-runtime:0.0.0
+
+# Allow to choose right options according to cases conditions.
+# Parameter cases is the array of two dimensional array, which
+# consist of condition bool value and expected value, if this
+# condition is TRUE.
+# e.g.
+# switch
+#   *
+#     TRUE
+#     "this value will be returned"
+#   *
+#     FALSE
+#     "this value will be skipped"
+#
+# This object returns value of only first truly statement.
+[cases...] > switch
+  if. > @
+    not.
+      eq.
+        cases.length
+        0
+    if.
+      gt.
+        length.
+          [] > arr
+            reduce. > @
+              cases
+              *
+              [acc case]
+                if. > @
+                  case.at 0
+                  acc.with (case.at 1)
+                  acc
+        0
+      arr.at 0
+      TRUE
+    "switch cases are empty" > msg
```

Comment:
@Graur I believe it's a typo. You forgot to make an abstract object here.

Reply:
@yegor256 Yep, it is, thanks!